content/SCALETutorials/Apps/CommunityApps/MinIOClustersSCALE/MigratingFromS3Service.md

```diff
@@ -0,0 +1,202 @@
+
+* A TrueNAS self-signed certificate for the MinIO **Enterprise** app (not required for the community **Charts** app).
+
+* New port number for the MinIO **Charts** app. 
+  Do not use the same port numbers as the S3 service (**9000** and **9001**). 
+  Refer to [Default Ports](https://www.truenas.com/docs/references/defaultports/) to identify available port numbers.
+  The MinIO **Enterprise** app can use the default port numbers **30000** and **30001**.
+
+* A MinIO Client (MC) release release that supports the commands and functions required to migrate from the S3 service to later MinIO releases. 
+  Download and install a MinIO Client (MC) within the **RELEASE.2022-06-26T18-51-48Z** and **RELEASE.2022-10-29T10-09-23Z** range. 
+
+## Migrating the MinIO S3 Service 
+This procedure uses the MinIO Client (MC), TrueNAS SCALE, and the MinIO web portal.
+
+1. Create a new [dataset]({{< relref "Datasetsscale.md" >}}) named **data** for the MinIO app. 
+   Do not use the same dataset(s) or path(s) consumed by the existing S3 service! 
+   For example, if the the existing S3 service path is */tank/s3/data*, create a new dataset and path */tank/minio/data* to use for the MinIO app.
+
+   You need at least double the storage capacity of the current S3 MinIO deployment on the TrueNAS SCALE system. 
+   Data storage should not cause the pool to exceed 80% utilization through the migration process. 
+
+   This tutorial uses the same system for the S3 service and the MinIO app deployment. 
+   Using a secondary system for storage capacity slows data transfer rates to and from the server.
+
+2. Identify two new port numbers to use for the community **Charts** MinIO app.
+   The community **Charts** MinIO app default ports are the same as the S3 service so select new ports to use. 
+   Do not use the existing MinIO S3 service ports (**9000** and **9001**). 
+   See [Default Ports](https://www.truenas.com/docs/references/defaultports/) for a list of assigned port numbers before deciding on the transitional app port numbers.
+   The **Enterprise** MinIO app uses ports **30000** and **30001**.
+   
+3. Download and install a MinIO Client release on your computer to facilitate the migration. 
+   Windows, Linux, and Mac version are available. 
+   Select a version that is between **RELEASE.2022-06-26T18-51-48Z** and **RELEASE.2022-10-29T10-09-23Z**. 
+   Versions outside of this range do not support the commands required to migrate the S3 service deployment to newer MinIO releases. 
+   Attempting to use a release outside the provided range can result in migration failure and the MinIO app failing to launch. 
+   This procedure uses MC **RELEASE.2022-10-29T10-09-23Z**. 
+
+   This procedure uses the Windows Subsystem for Linux (WSL) to install the client, and assumes user familiarity with WSL or standard Linux CLI.
+   Open a command-line tool such as PowerShell or Windows Command Prompt, then enter this command to install the MC client software:
```

Comment:
The following commands will not work properly if executed in PowerShell or Command Prompt, it must be executed in a Windows Subsystem for Linux CLI. 

If we do not have any articles for setting up WSL it may benefit us to link to Microsoft's WSL setup docs. https://learn.microsoft.com/en-us/windows/wsl/install

Reply:
I'll put a few links in for WSL.